kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/strings.kt

```diff
@@ -86,11 +87,30 @@ object StringShrinker : Shrinker<String> {
    }
 }
 
+/**
+ * Shrinks a string. Shrunk variants will be shorter (reduced in length, until they are [minLength] in length),
+ * and simplified (characters will be replaced by [simplestChar]).
+ *
+ * Note: by default [simplestChar] is the character "`a`". If shrinking an `Arb<String>` that does not include "`a`",
+ * either provide a suitable alternative, or use the secondary constructor and provide a suitable [Codepoint] [Arb],
+ * and a suitable character will be randomly selected.
+ */
 class StringShrinkerWithMin(
    private val minLength: Int = 0,
    private val simplestChar: Char = 'a',
```

Comment:
I think to be more clear and avoid accidental usage, the primary constructor with the `simplestChar` should be converted to a secondary constructor, and then `@Deprecated`.  The primary, and only, constructor should accept `codepoints: Arb<Codepoint>`, as it is more useful than a static character.

Thoughts?

Reply:
Do we want shrinks to be random? wouldn't it be better if a shrunk value is always deterministic ?

Reply:
Umm good question. I'm not sure, I hadn't thought about it and I don't have as good overview to answer it to be honest. My instinct would be that it's fine, and allowing for Arbs in shrinkers to make them random is okay. But maybe there's benefit to them being completely deterministic. Or there should be a principle that shrinkers should be deterministic.

If this PR doesn't fit then that's okay :)

Reply:
I like the idea but I think shrinks should be deterministic.
Like to be, 'a' is the best candidate for a-z shrinking. Perhaps there's a way to always choose the "best" shrink given a list of chars?

Reply:
How about this: The simplest char could be the first character of the pre-shrink String?

```kotlin
   override fun shrink(value: String): List<String> {

      val isShortest = value.length == minLength

      val simplestChar: Char? = value.firstOrNull()
      val isSimplest = simplestChar == null || value.all { it == simplestChar }
      // ...
```

Reply:
I've updated the PR to reflect my suggestion. I hope it's something that works! 

The old behaviour of a static 'simplest char' can still be retained, if necessary. I've written, but commented-out, an example secondary constructor with a `@Deprecated`.

It's not ready for merging as I haven't written tests, just updated the existing ones. In particular there's no testing for when `simplestCharSelector` returns `null`.

Reply:
Sorry for delay in replying.

I think that the simplest char being the first char in the input string could work.
It would be repeatable for any given input at least.

Another option is to introduce some kind of Codepoints/Alphabet interface which specifies the low point for a given range of chars ?

Reply:
No problem! I hope you had a good time over Christmas and New Years.

I'm not quite sure how an `Alphabet` interface would work, it would help to see an example. 

How should the 'simplest' char be selected? Would users know what this would mean? Would the 'simplest char' be relevant to anything else, or only `StringShrinker`? If 'simplest char' is only relevant to `StringShrinker` then it should only be defined and used there, so the context is clearer.

What happens if an `Alphabet` needs to be filtered or mapped - would the simplest char need to be defined on each operation? 

When defining a 'simplest char', does it have to be contained within the `Alphabet`? What happens if it isn't?

Reply:
@sksamuel  I've added the tests for `simpleCharSelector`. Aside from the deprecation annotations and backwards compatibility, I think it's good to go! Let me know if there's anything more I can do.